IbisHardwareIGSIO/ibishardwareIGSIO.cpp

```diff
@@ -386,23 +388,51 @@ void IbisHardwareIGSIO::InitPlugin()
 {
     m_plusConfigDirectory = GetIbisAPI()->GetConfigDirectory() + QString("PlusToolkit/");
     m_ibisPlusConfigFilesDirectory = m_plusConfigDirectory + QString("IbisPlusConfigFiles/");
-    m_plusToolkitPathsFilename = m_plusConfigDirectory + QString("plus-toolkit-paths.txt");
     m_plusConfigFilesDirectory = m_plusConfigDirectory + QString("PlusConfigFiles/");
 
-    // Look for the Plus Toolkit path
-    if( QFile::exists( m_plusToolkitPathsFilename ) )
+    // Look for the Plus Toolkit path and Plus Server executable
+    // Full path of the executable of plus server
+    QString plusServerExec = GetIbisAPI()->GetCustomPath( PlusServerExecutable );
+    bool ok = true;
+    if( plusServerExec != QString::null && !plusServerExec.isEmpty() )
     {
-        QFile pathFile( m_plusToolkitPathsFilename );
-        pathFile.open( QIODevice::ReadOnly | QIODevice::Text );
-        QTextStream pathIn( &pathFile );
-        pathIn >> m_plusServerExec;
+        QFileInfo fi( plusServerExec );
+        if( !( fi.isFile() && fi.isExecutable() ) )
+        {
+            ok = false;
+        }
+    }
+    else
+        ok = false;
+    if( !ok )
+    {
+        GetIbisAPI()->UnRegisterCustomPath( PlusServerExecutable );
+        GetIbisAPI()->RegisterCustomPath( PlusServerExecutable, "" );
     }
 }
 
-bool IbisHardwareIGSIO::LauchLocalServer( int port, QString plusConfigFile )
+bool IbisHardwareIGSIO::LaunchLocalServer( int port, QString plusConfigFile )
 {
+    bool ok = true;
+    QString plusServerExec = GetIbisAPI()->GetCustomPath( PlusServerExecutable );
+    if( plusServerExec != QString::null && !plusServerExec.isEmpty() )
+    {
+        QFileInfo fi( plusServerExec );
+            if( !( fi.isFile() && fi.isExecutable() ) )
+                ok = false;
+    }
+    else
+        ok = false;
+    if( !ok )
+    {
+        QString message;
+        message = QString("PlusServer not found.\n");
+        message = QString("Go to Settings/Preferences and set PlusServer executable.");
+        QMessageBox::warning(0, "Warning", message);
+        return false;
+    }
```

Comment:
This simpler piece of code would do the same:
```
    QString plusServerExec = GetIbisAPI()->GetCustomPath( PlusServerExecutable );
    if( plusServerExec != QString::null && !plusServerExec.isEmpty() )
    {
        QString message;
        message = QString("PlusServer executable path not defined.\n");
        message = QString("Go to Settings/Preferences and set PlusServer executable.");
        QMessageBox::warning(0, "Error", message);
        return false;
    }
    QFileInfo fi( plusServerExec );
    if( !( fi.isFile() && fi.isExecutable() ) )
    {
        QString message = QString("Can't execute PlusServer: %1").arg(plusServerExec);
        QMessageBox::warning(0,"Error", message );        
        return false;
    }
```

Reply:
I'll change this.

Reply:
Should I disallow removing paths in preference dialog. If user can do it, than I have to re-register path in Launch....